boot/freeldr/freeldr/arch/realmode/helpers_all.inc

```diff
@@ -0,0 +1,81 @@
+/*
+ * PROJECT:     FreeLoader
+ * LICENSE:     MIT (https://spdx.org/licenses/MIT)
+ * PURPOSE:     Real mode helper code for all architectures
+ * COPYRIGHT:   Copyright 2024 Daniel Victor <ilauncherdeveloper@gmail.com>
+ */
+
+Relocator16SS:
+    .word 0
+Relocator16EAX:
+    .long 0
+Relocator16ESP:
+    .long 0
+
+cs_writestr:
+    push ax
+    push ds
+    mov ax, cs
+    mov ds, ax
+
+    call writestr
+    pop ds
+    pop ax
+    ret
+
+GetCurrentBaseAddress:
+    /* Clear necessary registers */
+    xor eax, eax
+    xor ebx, ebx
+    xor ecx, ecx
+
+    /* Call the sub function to get the base address */
+    call .SubCall
```

Comment:
This also will return to the caller of `GetCurrentBaseAddress` ?

Reply:
no it will return to the function before `UpdateCurrentCodeSegment`.